mnist_igor_two.py

```diff
@@ -0,0 +1,254 @@
+
+    agent_compute_time = 0
+    agent_observe = True
+    start_time = time.time()
+    reset_time = time.time()
+    gain = 0
+    reward = None
+
+    # the first to start from
+    index = 0
+    graph = NetworkStatusGraph(cells=[c for c in agent.cells if not "mot" in c.name])
+    graph.plot()
+
+    # %%
+    correct_arr = np.zeros(MNIST_LABELS, dtype=int)
+    predict_arr = np.zeros(MNIST_LABELS, dtype=int)
+    processed = 0
+    # info lists of last truue and predicted;
+    last_true = []
+    last_predicted = []
+    while True:
+        # info read the current input
+        y = y_train[index]
+        obs = x_train[index]
+
+        # write time before agent step
+        current_time_relative = (time.time() - agent_compute_time)
+        # warning what do we need stepsize for?
+        if agent_compute_time > 0:
+            stepsize = current_time_relative * 1000
+        else:
+            stepsize = None
+
+        # info compute the agent activation
+        output = agent.step(observation=obs, output_type="rate", sort_func=lambda x: -x.value)[0]
+
+        # info get the predicted value and compare with the correct one
+        predicted = -1
+        if output.value > -1:
+            predicted = output.index
+            predict_arr[predicted] += 1
+
+        last_true.append(y)
+        last_predicted.append(predicted)
+        if predicted == y:
+            reward = 1
+            correct_arr[predicted] += 1
+            print("{:05d}: recognized {:d}\t".format(processed, y), correct_arr, "/", predict_arr,
+                  "\t({:.3f}%)".format(np.sum(correct_arr) / (processed + 1)))
+        else:
+            reward = -1
+        # info reward the agent accordingly to the output
+        # todo look carefully inside
+        agent.make_reward_step(reward=reward)
+
+        # write time after agent step
+        agent_compute_time = time.time()
+
+        last_predicted = last_predicted[-display_interval:]
+        last_true = last_true[-display_interval:]
+        if processed > 0 and processed % display_interval == 0:
+            # info update input image image
+            # obj.set_data(obs)
+            # ax.set_title('Predicted: %s True: %s' % (predicted, y))
+            # info update weights in graph
+            # graph.update_spikes(agent.sim.t)
+            # graph.update_weights('w')
+            # info update heatmap
+            # hitmap_graph.plot()
+            plt.draw()
+            plt.pause(1e-9)
+            # agent.rec_input.plot(animate=True, position=(4, 4))
+            # info display output act for last display_interval examples
+            # info left AGENT_STEPSIZE / DT for additional steps on the left of display
+            agent.rec_output.plot(animate=True,
+                                  steps=int(AGENT_STEPSIZE / DT + 2 * display_interval * AGENT_STEPSIZE / DT),
+                                  true_class=last_true, pred_class=last_predicted, stepsize=AGENT_STEPSIZE, dt=DT)
```

Comment:
This code requires changes to neuron++ so before we merge them - I can't merge this one to the gym repo